In [1]:
import pandas as pd
import numpy as np

In [2]:
# Zadanie 1a
pensje = pd.Series({
    'Anna': 5500, 
    'Jan': 7200, 
    'Ewa': 4800, 
    'Marek': 9100, 
    'Kasia': 6300
})

# Zadanie 1b
print(f"Osoba z najwyższą pensją: {pensje.idxmax()}")
print(f"Średnia pensja: {pensje.mean():.2f} zł")
print(f"Osoby powyżej średniej:\n{pensje[pensje > pensje.mean()]}")
print(f"Liczba osób zarabiających poniżej 6000: {(pensje < 6000).sum()}")

# Zadanie 1c
nowe_pensje = (pensje * 1.10) + 500
print(f"\nRanking nowych pensji:\n{nowe_pensje.sort_values(ascending=False)}")

Osoba z najwyższą pensją: Marek
Średnia pensja: 6580.00 zł
Osoby powyżej średniej:
Jan      7200
Marek    9100
dtype: int64
Liczba osób zarabiających poniżej 6000: 2

Ranking nowych pensji:
Marek    10510.0
Jan       8420.0
Kasia     7430.0
Anna      6550.0
Ewa       5780.0
dtype: float64


In [3]:
# Zadanie 2a i 2b
sklep = pd.DataFrame({
    'produkt': ['Laptop', 'Tablet', 'Smartfon', 'Słuchawki', 'Monitor'],
    'cena': [3500, 1800, 2500, 350, 2200],
    'stan_magazynowy': [45, 120, 200, 500, 75],
    'kategoria': ['Komputery', 'Mobilne', 'Mobilne', 'Akcesoria', 'Komputery']
})

sklep['wartosc_magazynu'] = sklep['cena'] * sklep['stan_magazynowy']
sklep['cena_z_vat'] = (sklep['cena'] * 1.23).round(2)
display(sklep) # display() wygląda ładniej w notebooku niż print()

# Zadanie 2c
url_tips = 'https://raw.githubusercontent.com/mwaskom/seaborn-data/master/tips.csv'
tips = pd.read_csv(url_tips)
print(f"\nKształt tips: {tips.shape}")
display(tips.sample(3))

,produkt,cena,stan_magazynowy,kategoria,wartosc_magazynu,cena_z_vat
0,Laptop,3500,45,Komputery,157500,4305.0
1,Tablet,1800,120,Mobilne,216000,2214.0
2,Smartfon,2500,200,Mobilne,500000,3075.0
3,Słuchawki,350,500,Akcesoria,175000,430.5
4,Monitor,2200,75,Komputery,165000,2706.0



Kształt tips: (244, 7)


,total_bill,tip,sex,smoker,day,time,size
82,10.07,1.83,Female,No,Thur,Lunch,1
215,12.90,1.10,Female,Yes,Sat,Dinner,2
143,27.05,5.00,Female,No,Thur,Lunch,6


In [4]:
url_penguins = 'https://raw.githubusercontent.com/mwaskom/seaborn-data/master/penguins.csv'
penguins = pd.read_csv(url_penguins)

# Zadanie 3a, 3b, 3c, 3d
print("--- INFO ---")
penguins.info()

print("\n--- STATYSTYKI ---")
display(penguins.describe())
print(f"Mediana masy ciała: {penguins['body_mass_g'].median()} g")

print("\n--- ROZKŁADY ---")
print(penguins['species'].value_counts())
print(f"\nProcent gatunku Adelie: {penguins['species'].value_counts(normalize=True)['Adelie']:.1%}")

print("\n--- BRAKI DANYCH ---")
print(penguins.isna().sum())

--- INFO ---
<class 'pandas.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 7 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    str    
 1   island             344 non-null    str    
 2   bill_length_mm     342 non-null    float64
 3   bill_depth_mm      342 non-null    float64
 4   flipper_length_mm  342 non-null    float64
 5   body_mass_g        342 non-null    float64
 6   sex                333 non-null    str    
dtypes: float64(4), str(3)
memory usage: 18.9 KB

--- STATYSTYKI ---


,bill_length_mm,bill_depth_mm,flipper_length_mm,body_mass_g
count,342.000000,342.000000,342.000000,342.000000
mean,43.921930,17.151170,200.915205,4201.754386
std,5.459584,1.974793,14.061714,801.954536
min,32.100000,13.100000,172.000000,2700.000000
25%,39.225000,15.600000,190.000000,3550.000000
50%,44.450000,17.300000,197.000000,4050.000000
75%,48.500000,18.700000,213.000000,4750.000000
max,59.600000,21.500000,231.000000,6300.000000


Mediana masy ciała: 4050.0 g

--- ROZKŁADY ---
species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

Procent gatunku Adelie: 44.2%

--- BRAKI DANYCH ---
species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
dtype: int64


In [5]:
tips['tip_pct'] = (tips['tip'] / tips['total_bill'] * 100).round(1)

print(f"Średni rachunek: {tips['total_bill'].mean():.2f}$")
print(f"Średni napiwek (%): {tips['tip_pct'].mean():.2f}%")
print(f"Najczęstszy dzień: {tips['day'].value_counts().idxmax()}")
print(f"Liczba palaczy: {tips['smoker'].value_counts()['Yes']}")
print(f"Mediana napiwku: {tips['tip'].median()}$")
print(f"Udział obiadów (Dinner): {tips['time'].value_counts(normalize=True)['Dinner']:.1%}")

Średni rachunek: 19.79$
Średni napiwek (%): 16.08%
Najczęstszy dzień: Sat
Liczba palaczy: 93
Mediana napiwku: 2.9$
Udział obiadów (Dinner): 72.1%


### Wnioski z analizy datasetu Penguins:
1. Dataset zawiera **344** pingwiny opisane przez **7** cech.
2. Najliczniejszym gatunkiem jest **Adelie** (152 osobniki, co stanowi ok. 44% populacji).
3. Średnia masa ciała pingwina to ok. **4201 g**, ale mediana wynosi **4050 g** (połowa pingwinów waży mniej).
4. Najwięcej pingwinów zamieszkuje wyspę **Biscoe** (168 osobników).
5. Dane są niekompletne – w kolumnie `sex` brakuje **11** wpisów (ok. 3.2% danych).